storage/client/blob/src/samples/java/blob/BasicExample.java

```diff
@@ -0,0 +1,132 @@
+         * The URL typically looks like this:
+         */
+        String endpoint = String.format(Locale.ROOT, "https://%s.blob.core.windows.net", accountName);
+
+        /**
+         * Create a StorageClient object that wraps the service endpoint, credential and a request pipeline.
+         */
+        StorageClient storageClient = StorageClient.storageClientBuilder().endpoint(endpoint).credential(credential).buildClient();
+
+        /**
+         * This example shows several common operations just to get you started.
+         */
+
+        /**
+         * Create a client that references a to-be-created container in your Azure Storage account. This returns a
+         * ContainerClient object that wraps the container's endpoint, credential and a request pipeline (inherited from storageClient).
+         * Note that container names require lowercase.
+         */
+        ContainerClient containerClient = storageClient.getContainerClient("myjavacontainerbasic" + System.currentTimeMillis());
+
+        /**
+         * Create a container in Storage blob account.
+         */
+        containerClient.create();
+
+        /**
+         * Create a client that references a to-be-created blob in your Azure Storage account's container.
+         * This returns a BlockBlobClient object that wraps the blob's endpoint, credential and a request pipeline
+         * (inherited from containerClient). Note that blob names can be mixed case.
+         */
+        BlockBlobClient blobClient = containerClient.getBlockBlobClient("HelloWorld.txt");
+
+        String data = "Hello world!";
+        InputStream dataStream = new ByteArrayInputStream(data.getBytes());
+
+        /**
+         * Create the blob with string (plain text) content.
+         */
+        blobClient.upload(dataStream, data.length());
+
+        dataStream.close();
+
+        /**
+         * Download the blob's content to output stream.
+         */
+        OutputStream outputStream = new ByteArrayOutputStream(data.length());
+        blobClient.download(outputStream);
+        outputStream.close();
+
+        /**
+         * Verify that the blob data round-tripped correctly.
+         */
+        if (!data.equals(outputStream.toString())) {
+            throw new RuntimeException("The downloaded data does not match the uploaded data.");
+        }
+
+        /**
+         * Create more blobs before listing.
+         */
+        for(int i = 0; i < 3; i++) {
+            String sampleData = "Samples";
+            InputStream dataInBlobs = new ByteArrayInputStream(sampleData.getBytes(Charset.defaultCharset()));
+            containerClient.getBlockBlobClient("myblobsforlisting" + System.currentTimeMillis())
+                    .upload(dataInBlobs, sampleData.length());
+            dataInBlobs.close();
+        }
+
+        /**
+         * List the blob(s) in our container.
+         */
+        containerClient.listBlobsFlat(new ListBlobsOptions().maxResults(1), null)
```

Comment:
Using maxResults here is a bit misleading, that field will determine the number of results a single page will return. I think this should be removed.

Reply:
We really need to rename max results. It's not intuitive that it limits the items per page of response. Not necessary for this review; I'm tracking that discussion elsewhere.

Reply:
Done. Removed for now.